auth_with_external_browser_test.go

```diff
@@ -0,0 +1,189 @@
+package gosnowflake
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestExternalBrowserSuccessful(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Success, cfg.User, cfg.Password)
+	}()
+	go func() {
+		defer wg.Done()
+		err := connectToSnowflake(cfg, "SELECT 1", true)
+		assertNilF(t, err, fmt.Sprintf("Connection failed due to %v", err))
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserFailed(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	cfg.ExternalBrowserTimeout = time.Duration(10) * time.Second
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Fail, "FakeAccount", "NotARealPassword")
+	}()
+	go func() {
+		defer wg.Done()
+		tOut := "authentication timed out"
+		err := connectToSnowflake(cfg, "SELECT 1", false)
+		assertTrueF(t, err.Error() == tOut, fmt.Sprintf("Expected %v, but got %v", tOut, err))
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserTimeout(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	cfg.ExternalBrowserTimeout = time.Duration(1) * time.Second
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Timeout, cfg.User, cfg.Password)
+	}()
+	go func() {
+		defer wg.Done()
+		tOut := "authentication timed out"
+		err := connectToSnowflake(cfg, "SELECT 1", false)
+		assertTrueF(t, err.Error() == tOut, fmt.Sprintf("Expected %v, but got %v", tOut, err))
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserMismatchUser(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	correctUsername := cfg.User
+	cfg.User = "fakeAccount"
+	var wg sync.WaitGroup
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Success, correctUsername, cfg.Password)
+	}()
+	go func() {
+		defer wg.Done()
+		expectedErrorMsg := "390191 (08004): The user you were trying to authenticate " +
+			"as differs from the user currently logged in at the IDP."
+		err := connectToSnowflake(cfg, "SELECT 1", false)
+		assertTrueF(t, err.Error() == expectedErrorMsg, fmt.Sprintf("Expected %v, but got %v", expectedErrorMsg, err))
+	}()
+	wg.Wait()
+}
+
+func TestClientStoreCredentials(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	cfg.ClientStoreTemporaryCredential = 1
+	cfg.ExternalBrowserTimeout = time.Duration(10) * time.Second
```

Comment:
Can we reduce the timeouts? How much time it takes to auth correctly using your fake browser?